### PR TITLE
OCLOMRS-309: Implement edit internal mapping functionality

### PR DESCRIPTION
**Pull request by abulojoshua1**
# JIRA TICKET NAME:
[Implement edit internal mapping functionality](https://issues.openmrs.org/browse/OCLOMRS-309)

# Summary:
Enable editing of internal concept mappings.

### CODE REVIEW

File: src/components/dictionaryConcepts/components/MappingModal.jsx [hunk 1]
```diff
@@ -7,15 +7,17 @@ import {
...
     this.state = {
       type: 'Internal Mapping',
-      map_type: '',
+      to_concept_url: props.to_concept_url,
+      url: props.url,
+      map_type: props.map_type,
       concept_url: '',
       term: '',
-      source_url: '',
+      source_url: props.to_concept_url,
```
Comment (dessHub) [inline src/components/dictionaryConcepts/components/MappingModal.jsx]: You have used `props.*` more than two times, it make sense if you destructure it.

File: src/components/dictionaryConcepts/components/MappingModal.jsx [hunk 2]
```diff
@@ -37,11 +39,20 @@ class MappingModal extends Component {
     });
   };
 
+  submitMapping = () => {
+    const { to_concept_url, url, map_type } = this.state;
+    const { source } = this.props;
+    this.props.editMapping(url, { to_concept_url, map_type }, source);
```
Comment (Duncanian) [inline src/components/dictionaryConcepts/components/MappingModal.jsx]: Consider destructuring this prop

Comment (abulojoshua1) [reply src/components/dictionaryConcepts/components/MappingModal.jsx]: Ok, I will do just that.

File: src/components/dictionaryConcepts/components/MappingModal.jsx (hunk 1 above)
Comment (abulojoshua1) [reply src/components/dictionaryConcepts/components/MappingModal.jsx]: Ok, I will do just that, thanks!

File: src/tests/dictionaryConcepts/components/InternalMapping.test.js [hunk 3]
```diff
@@ -18,6 +26,24 @@ describe('render MappingModal', () => {
     expect(wrapper.length).toEqual(1);
   });
 
+  it('should render without breaking', () => {
```
Comment (dkayiwa) [inline src/tests/dictionaryConcepts/components/InternalMapping.test.js]: What should render without breaking?

Comment (abulojoshua1) [reply src/tests/dictionaryConcepts/components/InternalMapping.test.js]: Thanks for the keen eye, I will make it more descriptive.

**Pull request merged**